agent/internal/task_config.go

```diff
@@ -259,6 +259,9 @@ func (tc *TaskConfig) TaskAttributeMap() map[string]string {
 		evergreen.DistroIDOtelAttribute:           tc.Task.DistroId,
 		evergreen.VersionDescriptionOtelAttribute: tc.PatchOrVersionDescription,
 	}
+	if tc.Task.DisplayTaskDisplayName != nil && *tc.Task.DisplayTaskDisplayName != "" {
```

Comment:
nit: can be simplified down to `utility.FromStringPtr(tc.Task.DisplayTaskDisplayName) != ""`